src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.UnitTests/Application/AddressLookup/AddressLookupQueryHandlerTests.cs

```diff
@@ -0,0 +1,33 @@
+﻿using AutoFixture.NUnit3;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NUnit.Framework;
+using SFA.DAS.RoatpCourseManagement.Application.AddressLookup.Queries;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.RoatpCourseManagement.UnitTests.Application.AddressLookup
+{
+    [TestFixture]
+    public class AddressLookupQueryHandlerTests
+    {
+        [Test, AutoData]
+        public async Task Handle_CallsLocationService(GetAddressesListResponse data)
+        {
+            var postcode = "CV1 1ET";
+            var expectedAddresses = data.Addresses.Select(a => (AddressItem)a).ToArray();
+            var mockService = new Mock<ILocationLookupService>();
+            mockService.Setup(m => m.GetExactMatchAddresses(postcode)).ReturnsAsync(data);
+            var sut = new AddressLookupQueryHandler(mockService.Object, Mock.Of<ILogger<AddressLookupQueryHandler>>());
+
+            var result = await sut.Handle(new AddresssLookupQuery(postcode), new CancellationToken());
+
+            mockService.VerifyAll();
+            result.Addresses.Should().BeEquivalentTo(expectedAddresses);
+        }
+    }
```

Comment:
Add a test for null result please

Reply:
The result is never null as long as the API call is a success. If there are no results, the list is empty.

Reply:
Looking deeper, the service is using the dreaded Get method which supresses bad response. I will handle null.

Reply:
@MarkFCain I have updated the service to throw exception in case the response was not a success. Also the service returns null if the postcode does not match the regex, for this the handler just forwards null response to the controller which in turn returns a bad request.